app/models/theme_settings_migration.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+class ThemeSettingsMigration < ActiveRecord::Base
+  belongs_to :theme
+  belongs_to :theme_field
+
+  def calculate_diff(settings_before, settings_after)
+    diff = { additions: [], deletions: [] }
+
+    before_keys = settings_before.keys
+    after_keys = settings_after.keys
+
+    removed_keys = before_keys - after_keys
+    removed_keys.each { |key| diff[:deletions] << { key: key, val: settings_before[key] } }
+
+    added_keys = after_keys - before_keys
+    added_keys.each { |key| diff[:additions] << { key: key, val: settings_after[key] } }
+
+    common_keys = before_keys & after_keys
+    common_keys.each do |key|
+      if settings_before[key] != settings_after[key]
+        diff[:deletions] << { key: key, val: settings_before[key] }
+        diff[:additions] << { key: key, val: settings_after[key] }
+      end
+    end
+
+    self.diff = diff
+  end
+end
+
+# == Schema Information
+#
+# Table name: theme_settings_migrations
+#
+#  id             :bigint           not null, primary key
+#  theme_id       :integer          not null
```

Comment:
Do we need `theme_id` when we already have `theme_field_id`? I suppose if we want to fetch the `theme_id` we can just go through the `theme_fields` table.

Reply:
oooo ok I see this is being used as part of the unique index.